.github/workflows/checks.yml

```diff
@@ -424,3 +424,132 @@ jobs:
+          numbers=()
+          for num in "${raw_numbers[@]}"; do
+            trimmed_num=$(echo "$num" | tr -d '[:space:]')  # Remove all whitespace
+            numbers+=("$trimmed_num")
+          done
+
+          echo "Number of elements after splitting and trimming: ${#numbers[@]}"
+
+          # Check if there are exactly 8 numbers
+          if [ ${#numbers[@]} -ne 8 ]; then
+            echo "Error: EXPECTED_BITCOIN_DA_ID should contain exactly 8 numbers"
+            echo "Actual number of elements: ${#numbers[@]}"
+            exit 1
+          fi
+
+          # Check if all numbers are valid u32
+          for i in "${!numbers[@]}"; do
+            num=${numbers[$i]}
+            echo "Checking number $((i+1)): '$num'"
+            echo "Hex representation: $(echo -n "$num" | xxd -p)"
+            if ! [[ $num =~ ^[0-9]+$ ]]; then
+              echo "Error: '$num' is not composed of digits only"
+              exit 1
+            fi
+            if [ $num -gt 4294967295 ]; then
+              echo "Error: '$num' is greater than 4294967295"
+              exit 1
+            fi
+          done
+
+          # Reconstruct the trimmed DA_ID
+          trimmed_da_id=$(IFS=', '; echo "${numbers[*]}")
+
+          # Final check
+          if [ $comma_count -eq 7 ] && [ $space_count -eq 7 ] && [ ${#numbers[@]} -eq 8 ]; then
+            echo "EXPECTED_BITCOIN_DA_ID is valid:"
+            echo "- Contains 7 commas"
+            echo "- Contains 7 spaces"
+            echo "- Contains 8 valid u32 numbers"
+            echo "Original value: $EXPECTED_BITCOIN_DA_ID"
+            echo "Trimmed value: $trimmed_da_id"
+          else
+            echo "Error: EXPECTED_BITCOIN_DA_ID format is incorrect"
+            echo "- Comma count: $comma_count (should be 7)"
+            echo "- Space count: $space_count (should be 7)"
+            echo "- Number count: ${#numbers[@]} (should be 8)"
+            exit 1
+          fi
+
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Install Dependencies
+        run: |
+          sudo apt update && sudo apt -y install curl gcc cpp cmake clang llvm
+          sudo apt -y autoremove && sudo apt clean && sudo rm -rf /var/lib/apt/lists/*
+
+      - name: Install Rust
+        run: |
+          curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
+          rustup install 1.79.0
+          rustup default 1.79.0
+
+      - name: Install Cargo Binstall
+        run: |
+          source $HOME/.cargo/env
+          cargo install --version 1.7.0 cargo-binstall
+
+      - name: Install cargo-risczero
+        run: |
+          cargo binstall cargo-risczero@1.0.5 --no-confirm
+
+      - name: Install risc0-zkvm toolchain
+        run: cargo risczero install --version r0.1.79.0-2
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Build Project
+        env:
+          REPR_GUEST_BUILD: 1
+        run: |
+          cargo build --release
```

Comment:
that's gonna be some expensive check - 13 mins to check 1 string in the binary. Maybe move into nexttest?

Reply:
yeah makes sense